packages/@stimulus/core/src/scope.ts

```diff
@@ -40,7 +40,7 @@ export class Scope {
   }
 
   private queryElements(selector: string): Element[] {
-    return Array.from(this.element.querySelectorAll(selector))
+    return Array.from(this.element.querySelectorAll(CSS.escape(selector)))
```

Comment:
I don't think escaping the entire selector fixes the problem. In fact, it mostly breaks things. Here's a simplified example:
```js
> selector = `[data-controller~="hello"]`
"[data-controller~=\"hello\"]"

> escapedSelector = CSS.escape(selector)
"\\[data-controller\\~\\=\\\"hello\\\"\\]"

> document.querySelector(selector)
<div class="landing-hello" data-controller="hello">

> document.querySelector(escapedSelector)
null
```

`CSS.escape()` will need to be targeted more tightly within the selector.

Reply:
I could try to add CSS.espace only on the attribute names, where they are defined (targets, actions, controllers, ...). WDYT?

Reply:
I'd like the test suite help guide you. `yarn test` locally or, rebase this branch to pick up latest CI changes so you can see the test results.